pages/index.tsx

```diff
@@ -1,14 +1,45 @@
-import { FormEvent, useEffect, useState } from 'react';
+import { FormEvent, HTMLAttributes, useEffect, useState } from 'react';
 import { Button } from '~/components/Button';
 import { TitleText } from '~/components/TitleText';
 import { Input } from '../components/Input';
 import { DeleteModal } from '~/components/DeleteModal';
 import { EditModal } from '~/components/EditModal';
 import { createGift, getAllGifts } from '~/utils/apiRequests';
-import { Gift, CreateGift } from '~/shared/types';
+import { Gift, CreateGift, User } from '~/shared/types';
 import { handleGeneralError } from '~/utils/handleError';
+import {
+  GetServerSidePropsContext,
+  GetServerSidePropsResult,
+  InferGetServerSidePropsType,
+} from 'next';
+import { validateRequest } from '~/backend/auth';
+import axios from 'axios';
+import { useRouter } from 'next/router';
+import SvgUser from '~/icons/user';
+import SvgArrowRightStartOnRectangle from '~/icons/arrow_right_start_on_rectangle';
 
-export default function Home() {
+export async function getServerSideProps(
+  context: GetServerSidePropsContext,
+): Promise<GetServerSidePropsResult<{ user: User }>> {
+  const cookieData = await validateRequest(context.req, context.res);
+  if (!cookieData.user) {
+    return {
+      redirect: {
+        permanent: false,
+        destination: '/login',
+      },
+    };
+  }
+  return {
+    props: {
+      user: JSON.parse(JSON.stringify(cookieData.user)) as User,
+    },
+  };
+}
```

Comment:
En tiedä onko sulla kokemusta `Next`in `getServerSideProps`-funktiosta, että onko kuinka helppo saada geneeriseksi. Sain importit toimimaan toisesta tiedostosta, mutta en saanut `context`-argumenttia mitenkään siirrettyä sinne toiseen tiedostoon :smile: Lähinnä helpottaisi ettei tätä tarvitsisi jokaiseen tiedostoon kopioida :sweat_smile: (jos on mahdollista tehdä geneerinen)

Middlewareen kannattanee tehdä authentication-tsekkaus, mutta sitten jos authentication onkin kunnossa, niin serveriltä täytyy käyttäjän tiedot saada, tähän käytän `getServerSideProps`-funktiota

Reply:
Tällähän se lähti pelittämään! Kiitos jälleen todella paljon!! :)

Syy tosiaan sille, miksi en itse saanut toimimaan näyttäisi olevan se, että en koskaan exportannut `getServerSideProps`-funktiota

En ollut siis täysin tajunnut, mitä tällä `You can use getServerSideProps by exporting it from a Page Component.` tarkoitettiin, mutta nyt tiedän! :sweat_smile: :+1: